apps/system/js/notifications.js

```diff
@@ -426,9 +430,9 @@ var NotificationScreen = {
     notificationNode.dataset.type = type;
     notificationNode.dataset.manifestURL = manifestURL;
 
-    if (detail.icon) {
+    if (detail.appIcon) {
```

Comment:
In existing code "detail.icon" was never populated whenever a new notification arrived instead of that "detail.appIcon" getting populated.